megalinter/tests/test_megalinter/LinterTestRoot.py

```diff
@@ -38,3 +52,23 @@ def test_report_tap(self):
     def test_report_sarif(self):
         utilstest.linter_test_setup({"report_type": "SARIF"})
         utilstest.test_linter_report_sarif(self.get_linter_instance(), self)
+
+    def test_format_fix(self):
+        utilstest.linter_test_setup()
+
+        if self.linter_name == "prettier":
```

Comment:
I'd prefer to have linter-specific behaviors not in the core code, you could define a methode init_test_variables on linter root, that does nothing, and override it on PrettierLinter and StandardLinter ?

Reply:
@nvuillam I am still not clear on how to make the changes you are talking about because if you finally look at the changes I had to make in that file, it was in several tests and in the **test_format_fix** function in particular I had to add the changes very carefully because the order matters (changes between **linter_test_setup** and after **get_linter_instance**)....

Reply:
@nvuillam done!